cli/2024-06-04-aws-irsa-support.md

```diff
@@ -0,0 +1,376 @@
+
+
+## Objectives
+
+> Issue Reference: https://github.com/radius-project/radius/issues/7618 
+
+### Goals
+
+* Radius users can configure AWS provider and enable Radius to use IRSA for authentication and deployment of application.
+* IRSA can be configured via interactive experience.
+* IRSA can be configured with non-interactive commands.
+* Radius users can deploy and manage AWS resources without having to manage AWS access key id and secret
+
+### Non-goals
+
+* Azure Managed Identity support
+* Azure Workload Identity support
+
+The above two are addressed in [Azure Workload Identitu Support](https://github.com/radius-project/design-notes/blob/main/cli/2024-04-azure-workload-identity.md)
+
+* Ability of radified applications to be able to use IRSA to connect to AWS resources.
+
+This facilitates a user application to be able to configure its container to use IRSA to authenticate itself for communication with AWS. While the technology is IRSA, this feature is very different from enabling Radius to use IRSA to deploy resources.
+
+### User scenarios
+
+#### User Story 1
+
+As a user I should be able to configure Radius AWS provider with IRSA through interactive experience.
+
+```
+% rad init --full  
+Enter an environment name 
+>default 
+
+% rad init --full  
+Select your cloud provider                     
+1. Azure                                  
+> 2. AWS                                     
+3.[back] 
+
+% rad init --full  
+
+Select the identity for the AWS cloud provider 
+1. IAM access key                              
+> 2. IAM Role for Service Account (IRSA)
+
+% rad init --full  
+
+Enter the IAM RoleARN for configuring IRSA: <role-arn-radius>
+```
+
+
+#### User story 2
+
+As a user I should be able to configure Radius AWS provider with IRSA using non interactive commands.
+
+1. Install the `rad` CLI.
+
+2. Install Radius using `rad install kubernetes`
+    ```
+    rad install kubernetes --set global.awsIRSA.enabled=true #default false
+    ```
+    If enabled, it should print a success/ failure message after updating the pod spec to mount the token. It should also print a information message to the complete of configuration by using rad credential register and rad env update.
+
+3. Create a Radius environment and add an AWS cloud provider.
+    ```
+    rad group create default
+
+    rad env create default
+    
+    rad env update default --aws-account-id <aws-account-id> --aws-region <aws-region> 
+    ```
+
+4. Register the role ARN using `rad credential register aws irsa`.
+    ```
+   rad credential register aws irsa --iam-role <roleARN>
+    ```
+
+Radius is now configured to use AWS IRSA for authentication.
+
+## Design
+
+### High Level Design for IRSA
+
+During installation, Radius should allow user to enable IRSA for AWS while conifguring AWS provider. Then the user should be able to enter the roleARN associated with desired policies which will be stored as the AWS credential.
```

Comment:
It might be better stated as "users must choose whether or not to enable IRSA during Radius installation".